### PR TITLE
refactor: sidebar interaction by onclick

### PR DESCRIPTION
**Pull request by HeetVekariya**
# Fixes Issue:
- Issue: #748 
- Change Side Menu Behavior to Open on Click Instead of on Hover

<br>

# Changes made:
- Refactored code to interact with sidebar using `onClick` instead `onHover`

<br>

# Files changed:
- **menu.tsx** (frontend/webapp/components/side.menu/menu/menu.tsx)
- **menu.styled.tsx** (frontend/webapp/components/side.menu/menu/menu.styled.tsx)

<br>

- [x] Tested the code

### CODE REVIEW

File: frontend/webapp/components/side.menu/menu/menu.tsx [hunk 1]
```diff
@@ -76,15 +78,14 @@ export function Menu() {
...
+        <ContactUsButton expand={isExpanded} />
       </ContactUsWrapper>
     );
   }
 
   return (
     <MenuContainer
-      onMouseEnter={() => setIsHovered(true)}
-      onMouseLeave={() => setIsHovered(false)}
+      className={isExpanded ? 'expanded' : ''}
```
Comment (alonkeyval) [inline frontend/webapp/components/side.menu/menu/menu.tsx:88]: @HeetVekariya please send isExpanded as styled-component props to the width attribute to match the rest of the app

Comment (HeetVekariya) [reply frontend/webapp/components/side.menu/menu/menu.tsx]: Changes made

File: frontend/webapp/components/side.menu/menu/menu.tsx [hunk 2]
```diff
@@ -76,15 +78,14 @@ export function Menu() {
...
+        <ContactUsButton expand={isExpanded} />
       </ContactUsWrapper>
     );
   }
 
   return (
     <MenuContainer
-      onMouseEnter={() => setIsHovered(true)}
-      onMouseLeave={() => setIsHovered(false)}
+      className={isExpanded ? 'isExpanded' : ''}
```
Comment (alonkeyval) [inline frontend/webapp/components/side.menu/menu/menu.tsx:88]: please see instruction about passing props from styled-component docs: https://styled-components.com/docs/basics#passed-props

Comment (HeetVekariya) [reply frontend/webapp/components/side.menu/menu/menu.tsx]: changed the way prop is passed to the style.div according to docs

File: frontend/webapp/components/side.menu/menu/menu.styled.tsx [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
 
-export const MenuContainer = styled.div`
+export const MenuContainer = styled.div<{ $isExpanded?: boolean; }>`
   width: 70px;
   transition: width 0.1s;
   height: 100%;
   border-right: 1px solid rgba(255, 255, 255, 0.04);
   background: ${({ theme }) => theme.colors.dark_blue};
-  &:hover {
+  ${ props  => props.$isExpanded && `
```
Comment (alonkeyval) [inline frontend/webapp/components/side.menu/menu/menu.styled.tsx:9]: @HeetVekariya you can pass it directly to the width attr like this: 
`width: ${({$isExoanded}) => $isExoanded ? 234 : 70}px`

Comment (HeetVekariya) [reply frontend/webapp/components/side.menu/menu/menu.styled.tsx]: Made changes accordingly & tested the code

File: frontend/webapp/components/side.menu/menu/menu.styled.tsx [hunk 4]
```diff
@@ -1,15 +1,14 @@
...
+export const MenuContainer = styled.div<{ $isExpanded?: boolean; }>`
   width: 70px;
   transition: width 0.1s;
   height: 100%;
   border-right: 1px solid rgba(255, 255, 255, 0.04);
   background: ${({ theme }) => theme.colors.dark_blue};
-  &:hover {
-    width: 234px;
-  }
+  width: ${({$isExpanded}) => $isExpanded ? 234 : 70}px;
```
Comment (alonkeyval) [inline frontend/webapp/components/side.menu/menu/menu.styled.tsx:9]: @HeetVekariya  remove duplicate fields

Comment (HeetVekariya) [reply frontend/webapp/components/side.menu/menu/menu.styled.tsx:8]: done

File: frontend/webapp/components/side.menu/menu/menu.styled.tsx [hunk 5]
```diff
@@ -1,15 +1,13 @@
...
 `;
+
 export const LogoWrapper = styled.div`
   padding: 24px 16px;
   height: 48px;
```
Comment (alonkeyval) [inline frontend/webapp/components/side.menu/menu/menu.styled.tsx:13]: @HeetVekariya it looks good, we need to add here a pointer cursor attr to help users know that is logo is clickable

Comment (HeetVekariya) [reply frontend/webapp/components/side.menu/menu/menu.styled.tsx:14]: again done

**Review by alonkeyval — APPROVED** [review]
@HeetVekariya  LGTM

**Comment by HeetVekariya** [comment]
@alonkeyval Thank you for merging

**Pull request merged**